account_setup.py

```diff
@@ -0,0 +1,348 @@
+    """
+        Gives service principal contributor role authorization on scope
+        :param credentials: msrestazure.azure_active_directory.AdalAuthentication
+        :param subscription_id: str
+        :param scope: str
+        :param principal_id: str
+    """
+    authorization_client = AuthorizationManagementClient(credentials, subscription_id)
+    role_name = 'Contributor'
+    roles = list(authorization_client.role_definitions.list(
+        scope,
+        filter="roleName eq '{}'".format(role_name)
+    ))
+    contributor_role = roles[0]
+    for i in range(10):
+        try:
+            authorization_client.role_assignments.create(
+                scope,
+                uuid.uuid4(),
+                {
+                    'role_definition_id': contributor_role.id,
+                    'principal_id': principal_id
+                }
+            )
+            break
+        except CloudError as e:
+            # ignore error if service principal has not yet been created
+            time.sleep(1)
+            if i == 10:
+                raise e
+
+
+def format_secrets(**kwargs):
+    '''
+    Retuns the secrets for the created resources to be placed in secrets.yaml
+    The following form is returned:
+
+        service_principal:
+            tenant_id: <AAD Diretory ID>
+            client_id: <AAD App Application ID>
+            credential: <AAD App Password>
+            batch_account_resource_id: </batch/account/resource/id>
+            storage_account_resource_id: </storage/account/resource/id>
+    '''
+    return yaml.dump({"service_principal": kwargs}, default_flow_style=False)
+
+
+def prompt_for_confirmation(prompt, deny_error, unrecognized_input_error):
+    """
+        Prompt user for confirmation, 'y' for confirm, 'n' for deny
+        :param prompt: str
+        :param deny_error: Exception
+        :param unrecognized_input_error: Exception
+        :return None if prompt successful, else raises error
+    """
+    confirmation = input(prompt).lower()
+    for i in range(3):
+        if confirmation == "n":
+            raise deny_error
+        elif confirmation == "y":
+            break
+        elif confirmation != "y" and i == 2:
+            raise unrecognized_input_error
+        confirmation = input("Please input 'y' or 'n': ").lower()
+
+
+def prompt_with_default(key, value):
+    user_value = input("{0} [{1}]: ".format(key, value))
+    if user_value != "":
+        return user_value
+    else:
+        return value
+
+
+if __name__ == "__main__":
+    print("\nGetting credentials.")
+    # get credentials and tenant_id
+    creds, subscription_id = credentials.get_azure_cli_credentials()
+    subscription_client = SubscriptionClient(creds)
+    tenant_ids = [tenant.id for tenant in subscription_client.tenants.list()]
+    if len(tenant_ids) != 1:
+        raise AccountSetupError("More than one tenant configured on your subscription.")
```

Comment:
Is there nothing the user can do here? I'm not even sure what this error means myself...

Reply:
This means that you're subscription has more than one tenant configured (don't know myself why that is possible/desired). I'm not sure what should be done here, or if we should bother. This seems like an extremely advanced case.